app/controllers/StoreMyNinoController.scala

```diff
@@ -48,7 +50,27 @@ class StoreMyNinoController @Inject()(
   def onPageLoad: Action[AnyContent] = (authorisedAsFMNUser andThen getPersonDetailsAction) async {
     implicit request => {
       auditService.audit(AuditUtils.buildAuditEvent(request.personDetails, "ViewNinoLanding", frontendAppConfig.appName, None))
-      Future(Ok(view(request.nino.map(_.formatted).getOrElse(""))))
+      for {
```

Comment:
Do we need a 'for' here? A 'for' comprehension will execute sequentially. This is usually useful when you need the outcome of a call before executing the next line of code. It looks like each call here is not dependent on the other and could be executed in parallel.

Reply:
I have looked into this for you and got Ali’s opinion on your comment, we have determined that the for loop is only necessary because the google and apple ids return a future so we do not want to use an await.result to get their values.

Did you mean in your comment that the first id is generated then the next one in a sequence? If so, do you have any suggestions to possibly change this?

Reply:
Let me take a look. It may be an option to map or flatMap. For comprehensions are not just for returning futures.

Reply:
`  def onPageLoad: Action[AnyContent] = (authorisedAsFMNUser andThen getPersonDetailsAction) async {
    implicit request => {
      auditService.audit(AuditUtils.buildAuditEvent(request.personDetails, "ViewNinoLanding", frontendAppConfig.appName, None))

      val fullName: String = request.personDetails.person.fullName
      val nino: String     = request.nino.map(_.formatted).getOrElse("")

      storeMyNinoConnector.createGooglePass(fullName, nino).flatMap { googleId =>
        storeMyNinoConnector.createApplePass(fullName, nino).flatMap { appleId =>
          Future.successful(Ok(view(appleId.value, googleId.value, nino, isMobileDisplay(request))))
        }
      }
    }
  }`

Reply:
Parallel implementation 

`      val firstName = request.personDetails.person.fullName
      val nino      = request.nino.map(_.formatted).getOrElse("")
      val googleIdf = storeMyNinoConnector.createGooglePass(firstName, nino)
      val appleIdf  = storeMyNinoConnector.createApplePass(firstName, nino)

      googleIdf.flatMap {googleId =>
        appleIdf.map {appleId =>
          Ok(view(googleId.value, appleId.value, nino, isMobileDisplay(request)))
        }
      }`